backend/app/controllers/admin/users_controller.rb

```diff
@@ -42,5 +42,9 @@ class UsersController < Admin::ApplicationController
 
     # See https://administrate-prototype.herokuapp.com/customizing_controller_actions
     # for more information
+    def teacher
+      @teachers = User.where(role: 'teacher')
+      render json: @teachers
+    end
```

Comment:
since this `instance variable` `@teachers` will not be used in other places it should be a `local variable`